arangod/RocksDBEngine/RocksDBEngine.cpp

```diff
@@ -938,6 +938,25 @@ void RocksDBEngine::stop() {
     }
     _syncThread.reset();
   }
+ 
+  // wait for started compaction jobs to finish
+  bool compactionWarning = false;
+  while (true) {
+    {
+      READ_LOCKER(locker, _pendingCompactionsLock);
+      if (_runningCompactions == 0) {
+        break;
+      }
+    }
+      
+    if (!compactionWarning) {
+      LOG_TOPIC("9cbfd", INFO, Logger::ENGINES) << "waiting for compaction jobs to finish...";
+      compactionWarning = true;
+    }
+    // unfortunately there is not much we can do except waiting for
+    // RocksDB's compaction job(s) to finish.
+    std::this_thread::yield();
```

Comment:
I would rather use a sleep instead of a `yield`. The number of compaction jobs probably less than the number of available cores, so there is a good chance that there are no other threads waiting to be scheduled and the OS scheduler might immediately reschedule this thread...

Reply:
Ok, will change it.

Reply:
Done